lib/SIL/SILFunctionType.cpp

```diff
@@ -2628,9 +2628,17 @@ TypeConverter::getLoweredFormalTypes(SILDeclRef constant,
   SmallVector<TupleTypeElt, 4> inputs;
 
   // Merge inputs and generic parameters from the uncurry levels.
+  assert (uncurryLevel <= fnType.getParams().size());
   for (;;) {
     auto canInput = fnType->getInput()->getCanonicalType();
     auto inputFlags = ParameterTypeFlags().withInOut(isa<InOutType>(canInput));
+    if (!fnType.getParams().empty()) {
```

Comment:
This codepath needs to be migrated to speak in terms of `AnyFunctionType::Param`.  Simplify moving flags around from the first parameter isn't going to work in general.